include/engine/routing_algorithms.hpp

```diff
@@ -57,7 +57,16 @@ class RoutingAlgorithmsInterface
 template <typename Algorithm> class RoutingAlgorithms final : public RoutingAlgorithmsInterface
 {
   public:
-    RoutingAlgorithms(SearchEngineData<Algorithm> &heaps,
+    // HeapAlgorithm is an explicit heap algorithm type to be used with Algorithm
+    // - CH algorithms use CH heap
+    // - CoreCH algorithms use CH heap
+    // - MLD algorithms use MLD heap
+    using HeapAlgorithm = typename std::conditional<
+        std::is_same<Algorithm, routing_algorithms::corech::Algorithm>::value,
+        routing_algorithms::ch::Algorithm,
+        Algorithm>::type;
```

Comment:
Would `class SearchEngineData<CoreCH> : public SearchEngineData<CH> {}` work as well? That would enable us to remove the additional template parameter since we could use `SearchEngineData<Algorithm>`.